codex/blockexchange/engine/engine.nim

```diff
@@ -209,25 +211,26 @@ proc scheduleTasks(b: BlockExcEngine, blocks: seq[bt.Block]) =
   # schedule any new peers to provide blocks to
   for p in b.peers:
     for c in cids: # for each cid
-      # schedule a peer if it wants at least one
-      # cid and we have it in our local store
-      if c in p.peerWants and c in b.localStore:
-        if b.scheduleTask(p):
-          trace "Task scheduled for peer", peer = p.id
-        else:
-          trace "Unable to schedule task for peer", peer = p.id
-
-        break # do next peer
-
-proc resolveBlocks*(b: BlockExcEngine, blocks: seq[bt.Block]) =
+      # schedule a peer if it wants at least one cid
+      # and we have it in our local store
+      if c in p.peerWants:
+        if await (c in b.localStore):
```

Comment:
Wouldn't `if (await c) in b.localStore: ...` be more correct? I'm actually surprised this works at all.

Reply:
I wonder why you expect that. Check the full code: https://github.com/status-im/nim-codex/blob/e1f83a52b7bd68f10f0564a964de70de584f25a1/codex/blockexchange/engine/engine.nim#L217

`c` is just a cid, how we can await it? 

`c in b.localStore` refers to this operation: https://github.com/status-im/nim-codex/blob/e1f83a52b7bd68f10f0564a964de70de584f25a1/codex/stores/blockstore.nim#L57

So we just await a Future[bool]

Reply:
Ah, you're right, I wrongly assumed that `c` is a future.